src/shared/dissect-image.c

```diff
@@ -1452,6 +1452,17 @@ static int verity_partition(
                         if (!IN_SET(r, 0, -ENODEV, -ENOENT))
                                 return log_debug_errno(r, "Checking whether existing verity device %s can be reused failed: %m", node);
                         if (r == 0) {
+                                /* devmapper might say that the device exists, but the devlink might not yet have been
+                                 * created. Check and wait for the udev event in that case. */
+                                if (access(node, F_OK) != 0) {
```

Comment:
why this check? it should be safe to just call device_wait_for_devlink() unconditionally.

we really should try to make device_wait_for_devlink() robust so that it waits if the device is missing but is a NOP if it is already there, so that we never have to check anything beforehand.